### DISCUSSION TITLE
taro 如何实现小程序的动态渲染？

### DISCUSSION THREAD

**Discussion by cake221** [NervJS/taro#11840] [Q & A] [answered]
taro 提供了 jQuery 的调用方式，是怎么实现了小程序的动态渲染？

比如  `$("select").append("<view>视图</view>")` 这条命令。可以将 模版 `<view>视图</view>` 追加上到 taro构造的dom 上，后续是如何 增加到小程序的模版里。template 的写法，不应该提前写在 wxml 文件里么？js 如何追加到小程序上的呢？

**Comment by cake221** [comment]
感觉跟 taro 渲染 html 是一个原理。 https://taro-docs.jd.com/taro/docs/html/

**Comment by Chen-jj** [comment, marked as answer]
模板都在 dist/base.wxml 里提前生成好了

https://taro-docs.jd.com/taro/docs/implement-note#%E8%BF%90%E8%A1%8C%E6%97%B6

**Reply by cake221** [reply]
感谢回答。有点懂了。后面又看了看 wxParse 的实现，应该是一样的原理。
